hack/e2e.sh

```diff
@@ -18,7 +18,7 @@ function cleanup() {
         # Remove any testing labels that may have been added during tests
         echo "Removing any testing labels that were added to the cluster's nodes"
         nodes=( $(kubectl get nodes -l metering-node-testing-label="true" --no-headers | awk '{ print $1 }') )
-        for i in "${nodes[@]}"
+        for i in ${nodes[@]+"${nodes[@]}"}
```

Comment:
Also, it looks like we don't fail the e2e suite if we encounter a problem like this as we capture the exit code going into this cleanup function, and exit based on that same variable, which means an error un-labeling nodes or deleting testing namespaces does not influence the overall failure of the test/tests.

Reply:
Why do we cleanup anyway? Isn't the whole test namespace about to be deleted?

Reply:
Nodes are cluster-scoped and not namespace-scoped. We delete the testing namespaces in addition to the resources we may have created or modified during the last test iteration, which are a mixture of namespace (i.e. deleting the openshift-marketplace namespace is not a sufficient operation) and cluster scoped.